File: template/bit.py
N = 200_005
arr = [0 for i in range(N)]
diff = [0 for i in range(N)]


def lowbit(x):
    return x & (-x)


def add(x, v, n):
    i = x
    while i <= n:
        diff[i] += v
        i += lowbit(i)


def askSum(x):
    ans = 0
    i = x
    while i >= 1:
        ans += diff[i]
        i -= lowbit(i)
    return ans

File: template/bit.cpp
const int BIT_N = 200005;

struct
{
    int arr[BIT_N + 1] = {0};
    int diff[BIT_N + 1] = {0};

    void init(int n)
    {
        for (int i = 1; i <= n; ++i)
        {
            diff[i] = 0;
        }
    }

    int lowbit(int x)
    {
        return x & (-x);
    }

    void add(int x, int v, int n)
    {
        for (int i = x; i <= n; i += lowbit(i))
            diff[i] += v;
    }

    int ask(int x)
    {
        int ans = 0;
        for (int i = x; i >= 1; i -= lowbit(i))
            ans += diff[i];
        return ans;
    }

    int sumdigit(int x)
    {
        int ans = 0;
        while (x > 0)
        {
            ans += x % 10;
            x /= 10;
        }
        return ans;
    }
} bit;
